graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/processor/IfConditionModelBuilder.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.graphqlfirst.processor;
+
+import graphql.language.Directive;
+import graphql.language.InputValueDefinition;
+import io.stargate.db.query.Predicate;
+import java.util.Map;
+import java.util.Optional;
+
+class IfConditionModelBuilder extends ConditionModelBuilderBase<IfConditionModel> {
+
+  IfConditionModelBuilder(
+      InputValueDefinition argument,
+      String operationName,
+      EntityModel entity,
+      Map<String, EntityModel> entities,
+      ProcessingContext context) {
+    super(context, argument, operationName, entity, entities);
+  }
+
+  IfConditionModel build() throws SkipException {
+
+    Optional<Directive> ifDirective = DirectiveHelper.getDirective("cql_if", argument);
+    String fieldName =
+        ifDirective
+            .flatMap(d -> DirectiveHelper.getStringArgument(d, "field", context))
+            .orElse(argument.getName());
+    Predicate predicate =
+        ifDirective
+            .flatMap(d -> DirectiveHelper.getEnumArgument(d, "predicate", Predicate.class, context))
+            .orElse(Predicate.EQ);
```

Comment:
`build()` could be pulled up in `ConditionModelBuilderBase` if you introduce two abstract methods:
```java
protected abstract String getDirectiveName();
protected abstract void validate(FieldModel field, Predicate predicate) throws SkipException;
```

(also assuming that you use a single condition class per my other comment, so that the return statement at the end is the same)

Reply:
Nice idea! done